src/nyx/globals.h

```diff
@@ -32,7 +32,8 @@ namespace Nyxus
 
 	bool scanFilePairParallel(const std::string& intens_fpath, const std::string& label_fpath, int num_fastloader_threads, int num_sensemaker_threads, int filepair_index, int tot_num_filepairs);
 	std::string getPureFname(const std::string& fpath);
-	int processDataset(const std::vector<std::string>& intensFiles, const std::vector<std::string>& labelFiles, int numFastloaderThreads, int numSensemakerThreads, int numReduceThreads, int min_online_roi_size, bool save2csv, bool arrow_output, const std::string& csvOutputDir);
+	int processDataset(const std::vector<std::string>& intensFiles, const std::vector<std::string>& labelFiles, int numFastloaderThreads, int numSensemakerThreads, int numReduceThreads, int min_online_roi_size, bool save2csv, const std::string& csvOutputDir);
+	int processDataset(const std::vector<std::string>& intensFiles, const std::vector<std::string>& labelFiles, int numFastloaderThreads, int numSensemakerThreads, int numReduceThreads, int min_online_roi_size, const std::string& outputDir);
```

Comment:
Hard to guess what function fits a CSV, memory buffer, or Apache output usage. Could we have argument "save2csv" of enum type (values "save2csv", "save2buffer", "save2arrow", "save2parquet", etc) rather than bool or at least document these functions with a comment block ?

Reply:
Updated to use an enum to choose save type and added in empty classes to replace ifdef statements, as discussed with Samee. Let me know if you agree with this approach